lib/Support/Backedge.cpp

```diff
@@ -0,0 +1,41 @@
+//===- Backedge.cpp - Support classes for building backedges ----*- C++ -*-===//
```

Comment:
nit: no need for "*- C++ -*" in .cpp files.  emacs' little brain can handle the file suffix.

Reply:
I'd bet it also works for the 'hpp' file extension...